design/baremetal-operator/reboot-interface.md

```diff
@@ -138,6 +138,32 @@ The controller automatically removes all annotations with the
 
 - the Host is deprovisioned
 
+The API has been extended to permit either a hard or a soft reboot to
+take place, specified via the reboot annotation and a `mode` parameter. The API
+supports the following syntax:
+
+- `reboot.metal.io` -- immediate reboot via soft shutdown first,
+  followed by a hard shutdown if the soft shutdown fails.
+- `reboot.metal3.io: {'mode':'hard'}` -- immediate reboot via hard
+  shutdown, potentially allowing for high-availability use-cases.
+- `reboot.metal3.io/{key}` -- phased reboot, issued and managed by the
+  client registered with the `key`, via soft shutdown first, followed
+  by a hard reboot if the soft reboot fails.
+- `reboot.metal3.io/{key}: {'mode':'hard'}` -- phased reboot, issued
+  and managed by the client registered with the `key`, via hard
+  shutdown.
+
+The API will attempt to parse the `reboot.metal3.io` annotation value
```

Comment:
Then remove this paragraph. I don't think we need to emphasize the error handling, since that's implicit in describing what we expect to find in the map above.

Reply:
Done